admittance_controller/include/admittance_controller/admittance_rule.hpp

```diff
@@ -0,0 +1,224 @@
+     * 'current_joint_state'.
+     *
+     * \param[in] current_joint_state
+     * \param[in] measured_wrench
+     * \param[in] reference_joint_state
+     * \param[in] period
+     * \param[out] desired_joint_state
+     */
+  controller_interface::return_type update(
+    const trajectory_msgs::msg::JointTrajectoryPoint & current_joint_state,
+    const geometry_msgs::msg::Wrench & measured_wrench,
+    const trajectory_msgs::msg::JointTrajectoryPoint & reference_joint_state,
+    const rclcpp::Duration & period,
+    trajectory_msgs::msg::JointTrajectoryPoint & desired_joint_states);
+
+  /**
+     * Fill fields of `state_message` from current admittance controller state.
+     *
+     * \param[in] state_message
+     */
+  const control_msgs::msg::AdmittanceControllerState & get_controller_state();
+
+public:
+  // admittance config parameters
+  std::shared_ptr<admittance_controller::ParamListener> parameter_handler_;
+  admittance_controller::Params parameters_;
+
+protected:
+  /**
+     * Calculate the admittance rule from given the robot's current joint angles. the control frame rotation, the
+     * current force torque transform, the reference force torque transform, the force torque sensor frame id,
+     * the time delta, and the current admittance controller state. The admittance controller state input is updated
+     * with the new calculated values. A boolean value is returned indicating if any of the kinematics plugin calls
+     * failed.
+     * \param[in] current_joint_positions
+     * \param[in] base_control
+     * \param[in] base_ft
+     * \param[in] ref_base_ft
+     * \param[in] ft_sensor_frame
+     * \param[in] dt
+     * \param[in] admittance_state
+     * \param[out] success
+     */
+  bool calculate_admittance_rule(AdmittanceState & admittance_state, double dt);
+
+  /**
+     * Updates internal estimate of wrench in world frame `wrench_world_` given the new measurement. The `wrench_world_`
+     * estimate includes gravity compensation
+     * \param[in] measured_wrench
+     * \param[in] sensor_rot
+     * \param[in] cog_rot
+     */
+  void process_wrench_measurements(
+    const geometry_msgs::msg::Wrench & measured_wrench,
+    const Eigen::Matrix<double, 3, 3> & sensor_rot, const Eigen::Matrix<double, 3, 3> & cog_rot);
+
+  template <typename T1, typename T2>
+  void vec_to_eigen(const std::vector<T1> & data, T2 & matrix);
+
+  // number of robot joint
+  size_t num_joints_;
+
+  // Kinematics interface plugin loader
+  std::shared_ptr<pluginlib::ClassLoader<kinematics_interface::KinematicsInterface>>
+    kinematics_loader_;
+  std::unique_ptr<kinematics_interface::KinematicsInterface> kinematics_;
+
+  // filtered wrench in world frame
+  Eigen::Matrix<double, 6, 1> wrench_world_;
+
+  // admittance controllers internal state
+  AdmittanceState admittance_state_;
+
+  // transforms needed for admittance update
+  Transforms trans_;
+  Transforms trans_ref_;
```

Comment:
Changed variable name to `admittance_transforms_` moved all data to a single variable, instead of one for reference and one for current configuration.